### ISSUE TITLE
ReorderableWrap: adding scroll controller causes exception

### ISSUE THREAD

**Issue by GCamanes** [hanshengchiu/reorderables#146]
Hi @hanshengchiu,

First I need to say that your plugin is amazing and our team love it. Now we face a feature that need some control on the scrollview containing the ReorderableWrap widget.

## Exception log
The problem is, when we add a scroll controller, we got two exceptions when trying to drag an item.
```
======== Exception caught by gesture library =======================================================
The following assertion was thrown while routing a pointer event:
ScrollController not attached to any scroll views.
'package:flutter/src/widgets/scroll_controller.dart':
Failed assertion: line 107 pos 12: '_positions.isNotEmpty'


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack: 
#2      ScrollController.position (package:flutter/src/widgets/scroll_controller.dart:107:12)
#3      ScrollController.offset (package:flutter/src/widgets/scroll_controller.dart:115:24)
#4      _ReorderableWrapContentState._scrollTo (package:reorderables/src/widgets/reorderable_wrap.dart:549:51)
#5      _ReorderableWrapContentState._wrap.<anonymous closure>._onWillAccept (package:reorderables/src/widgets/reorderable_wrap.dart:1008:9)
#6      _ReorderableWrapContentState._wrap.<anonymous closure>.<anonymous closure> (package:reorderables/src/widgets/reorderable_wrap.dart:1025:42)
#7      _DragTargetState.didEnter (package:flutter/src/widgets/drag_target.dart:739:60)
#8      _DragAvatar.updateDrag.<anonymous closure> (package:flutter/src/widgets/drag_target.dart:895:23)
#9      ListMixin.firstWhere (dart:collection/list.dart:161:15)
#10     _DragAvatar.updateDrag (package:flutter/src/widgets/drag_target.dart:890:91)
#11     _DragAvatar.update (package:flutter/src/widgets/drag_target.dart:840:5)
#12     MultiDragPointerState._move (package:flutter/src/gestures/multidrag.dart:86:16)
#13     MultiDragGestureRecognizer._handleEvent (package:flutter/src/gestures/multidrag.dart:259:13)
#14     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:94:12)
#15     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:139:9)
#16     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:539:8)
#17     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:137:18)
#18     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:123:7)
#19     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:439:19)
#20     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:419:22)
#21     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:322:11)
#22     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
#23     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
#24     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:296:7)
#25     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:279:7)
#29     _invoke1 (dart:ui/hooks.dart:170:10)
#30     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:331:7)
#31     _dispatchPointerDataPacket (dart:ui/hooks.dart:94:31)
(elided 5 frames from class _AssertionError and dart:async)
====================================================================================================
```
```
======== Exception caught by gesture library =======================================================
The following assertion was thrown while routing a pointer event:
'package:flutter/src/widgets/drag_target.dart': Failed assertion: line 753 pos 12: '_candidateAvatars.contains(avatar) || _rejectedAvatars.contains(avatar)': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack: 
#2      _DragTargetState.didLeave (package:flutter/src/widgets/drag_target.dart:753:12)
#3      _DragAvatar._leaveAllEntered (package:flutter/src/widgets/drag_target.dart:925:26)
#4      _DragAvatar.finishDrag (package:flutter/src/widgets/drag_target.dart:936:5)
#5      _DragAvatar.end (package:flutter/src/widgets/drag_target.dart:848:5)
#6      MultiDragPointerState._up (package:flutter/src/gestures/multidrag.dart:151:14)
#7      MultiDragGestureRecognizer._handleEvent (package:flutter/src/gestures/multidrag.dart:263:13)
#8      PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:94:12)
#9      PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:139:9)
#10     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:539:8)
#11     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:137:18)
#12     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:123:7)
#13     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:439:19)
#14     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:419:22)
#15     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:322:11)
#16     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
#17     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
#18     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:296:7)
#19     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:279:7)
#23     _invoke1 (dart:ui/hooks.dart:170:10)
#24     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:331:7)
#25     _dispatchPointerDataPacket (dart:ui/hooks.dart:94:31)
(elided 5 frames from class _AssertionError and dart:async)
```

## Step to reproduce

1. flutter create testscroll
2. copy paste ReorderableWrap exemple to main page
3. add a scrollController to ReorderableWrap widget

## Flutter doctor
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.63.2)
[✓] Connected device (2 available)
[✓] HTTP Host Availability
```

**Comment by diegotori** [comment]
@GCamanes can you please create a reproducible example project so that I can better diagnose this issue? Thanks.

**Comment by GCamanes** [comment]
Hi @diegotori,

Thanks for answering me. Sure I can provide such a project to you.
Please find it [here](https://github.com/GCamanes/flutter-test-reorderables)

```
 % flutter doctor --verbose
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-x64, locale fr-FR)
    • Flutter version 2.10.3 at /Users/guillaumecamanes/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (3 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/guillaumecamanes/Library/Android/sdk
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = /Users/guillaumecamanes/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • iPhone 8 (mobile) • E8599860-007E-43DE-A657-A8D6A305EBEE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)      • chrome                               • web-javascript • Google Chrome 99.0.4844.74

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by aaumond** [comment]
Hi @diegotori, did you reproduce the problem mentioned by @GCamanes ?

**Comment by diegotori** [comment]
@aaumond Haven't had a chance to run it. Might take a stab at it this Sunday.

**Comment by diegotori** [comment]
@GCamanes managed to download your example project on my dev environment. Will attempt to run it in the next day or two. Sorry for the delay.

**Comment by diegotori** [comment]
@GCamanes after reviewing your example app, I was able to reproduce the issue. Turns out it was using the `ScrollController` without checking to see if there were any clients attached to it, which caused the exception.

As a result, `ReorderableWrap` now performs that check when calling `_scrollTo` and not scrolling. Hopefully the PR will get approved soon. Otherwise, I'll have to self-approve and deploy a new release tomorrow.
